snap/quota/quota.go

```diff
@@ -153,6 +221,253 @@ func (grp *Group) SliceFileName() string {
 	return buf.String()
 }
 
+// groupQuotaStats contains information about current quotas of a group
+// and is used by getGroupQuotaStats to contain this information
+type groupQuotaStats struct {
+	memoryLimit quantity.Size
+	memoryUsed  quantity.Size
+
+	cpuLimit int
+	cpuUsed  int
+
+	threadsLimit int
+	threadsUsed  int
+
+	allowedCPUsLimit []int
+	allowedCPUsUsed  []int
+}
+
+func max(a, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func maxq(a, b quantity.Size) quantity.Size {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+func (grp *Group) getAllowedCPUs() []int {
+	if grp.CPULimit == nil || len(grp.CPULimit.AllowedCPUs) == 0 {
+		return []int{}
+	}
+	return grp.CPULimit.AllowedCPUs
+}
+
+func (grp *Group) getTotalCPUPercentage() int {
+	if grp.CPULimit == nil || grp.CPULimit.Percentage == 0 {
+		return 0
+	}
+	return max(grp.CPULimit.Count, 1) * grp.CPULimit.Percentage
+}
+
+// getGroupQuotaStats Recursively retrieve current group quotas statistics, this should just
+// be invoked on the upper parent of a group tree, and then it will gather active quotas for the
+// tree and store them in the allQuotas paramater
+func (grp *Group) getGroupQuotaStats(allQuotas map[string]*groupQuotaStats, skipGrp *Group) *groupQuotaStats {
```

Comment:
this probably merits to be exported in export_test.go and get dedicated tests

Reply:
I have added better documentation and the unit test for the getGroupQuotaStats